src/video_core/renderer_vulkan/renderer_vulkan.cpp

```diff
@@ -969,18 +982,164 @@ void RendererVulkan::RenderScreenshot() {
+
+    // For a span of memory [x, x + s], import [AlignDown(x, alignment), AlignUp(x + s, alignment)]
+    // and maintain an offset to the start of the data
+    const u64 import_alignment = instance.GetMinImportedHostPointerAlignment();
+    const uintptr_t address = reinterpret_cast<uintptr_t>(settings.screenshot_bits);
+    void* aligned_pointer = reinterpret_cast<void*>(Common::AlignDown(address, import_alignment));
+    const u64 offset = address % import_alignment;
+    const u64 aligned_size = Common::AlignUp(offset + width * height * 4ull, import_alignment);
+
+    // Buffer<->Image mapping for the imported imported buffer
+    const vk::BufferImageCopy buffer_image_copy = {
+        .bufferOffset = offset,
+        .bufferRowLength = 0,
+        .bufferImageHeight = 0,
+        .imageSubresource =
+            {
+                .aspectMask = vk::ImageAspectFlagBits::eColor,
+                .mipLevel = 0,
+                .baseArrayLayer = 0,
+                .layerCount = 1,
+            },
+        .imageOffset = {0, 0, 0},
+        .imageExtent = {width, height, 1},
+    };
+
+    const vk::MemoryHostPointerPropertiesEXT import_properties =
+        device.getMemoryHostPointerPropertiesEXT(
+            vk::ExternalMemoryHandleTypeFlagBits::eHostAllocationEXT, aligned_pointer);
+
+    if (!import_properties.memoryTypeBits) {
+        // Could not import memory
+        return false;
+    }
+
+    vk::StructureChain<vk::MemoryAllocateInfo, vk::ImportMemoryHostPointerInfoEXT> allocation_chain;
+
+    const std::optional<u32> memory_type_index = FindMemoryType(
+        instance.GetPhysicalDevice().getMemoryProperties(),
+        vk::MemoryPropertyFlagBits::eHostVisible | vk::MemoryPropertyFlagBits::eHostCoherent,
+        import_properties.memoryTypeBits);
+
+    if (!memory_type_index.has_value()) {
+        // Could not find memory type index
+        return false;
+    }
+
+    auto& alloc_info = allocation_chain.get<vk::MemoryAllocateInfo>();
+    alloc_info.allocationSize = aligned_size;
+    alloc_info.memoryTypeIndex = memory_type_index.value();
+
+    auto& alloc_import_info = allocation_chain.get<vk::ImportMemoryHostPointerInfoEXT>();
+    alloc_import_info.pHostPointer = aligned_pointer;
+    alloc_import_info.handleType = vk::ExternalMemoryHandleTypeFlagBits::eHostAllocationEXT;
+
+    // Import host memory
+    const vk::UniqueDeviceMemory imported_memory =
+        device.allocateMemoryUnique(allocation_chain.get());
+
+    vk::StructureChain<vk::BufferCreateInfo, vk::ExternalMemoryBufferCreateInfo> BufferInfo;
+    auto& buffer_info = BufferInfo.get<vk::BufferCreateInfo>();
+    buffer_info.size = aligned_size;
+    buffer_info.usage = vk::BufferUsageFlagBits::eTransferDst;
+    buffer_info.sharingMode = vk::SharingMode::eExclusive;
+
+    auto& buffer_external_info = BufferInfo.get<vk::ExternalMemoryBufferCreateInfo>();
+    buffer_external_info.handleTypes = vk::ExternalMemoryHandleTypeFlagBits::eHostAllocationEXT;
```

Comment:
```suggestion
    const vk::StructureChain buffer_info = {
        vk::BufferCreateInfo{
            .size = aligned_size,
            .usage = vk::BufferUsageFlagBits::eTransferDst,
            .sharingMode = vk::SharingMode::eExclusive,
        },
        vk::ExternalMemoryBufferCreateInfo{
            .handleTypes = vk::ExternalMemoryHandleTypeFlagBits::eHostAllocationEXT,
        }
    };
```
ditto for structure chain above